FeedStoreChallenge/CoreDataFeedStore.swift

```
@@ -0,0 +1,198 @@
+//
+//  CoreDataFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by vinod supnekar on 22/01/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+enum CustomError : Error {
+	case noRecordFound
+	case fetchError
+}
+ 
+class CoreDataFeedStore: FeedStore {
+	
+	private struct CodableFeedImage: Equatable,Codable {
+		public let id: UUID
+		public let description: String?
+		public let location: String?
+		public let url: URL
+
+		init( image: LocalFeedImage) {
+			id = image.id
+			description = image.description
+			location = image.location
+			url = image.url
+		}
+		
+		public var local: LocalFeedImage {
+			return LocalFeedImage.init(id: id, description: description, location: location, url: url)
+			}
+	}
+	
+	private struct Cache: Codable {
+		let feed: [CodableFeedImage]
+		let timestamp: Date
+		
+		var localFeed: [LocalFeedImage] {
+			return feed.map { $0.local
+			}
+		}
+	}
+	
+	lazy var managedObjectModel: NSManagedObjectModel = {
+		let managedObjectModel = NSManagedObjectModel.mergedModel(from: [Bundle(for:  type(of: self))])!
```

Errors should be handled without `!` - if the model can't be found, you can throw an error so clients can handle it.